lib/service/Service.ts

```diff
@@ -55,9 +55,13 @@ const argChecks = {
   },
   NON_ZERO_QUANTITY: ({ quantity }: { quantity: number }) => { if (quantity === 0) throw errors.INVALID_ARGUMENT('quantity must not equal 0'); },
   PRICE_NON_NEGATIVE: ({ price }: { price: number }) => { if (price < 0) throw errors.INVALID_ARGUMENT('price cannot be negative'); },
-  SUPPORTED_CURRENCY: ({ currency }: { currency: string }) => {
-    const supportedCurrencies = ['BTC', 'LTC']; // TODO: detect supported currencies automatically instead of hardcoding
-    if (!supportedCurrencies.includes(currency)) throw errors.INVALID_ARGUMENT(`currency ${currency} is not supported`);
+  VALID_CURRENCY: ({ currency }: { currency: string }) => {
+    if (currency.length !== 3 || !currency.match(/^[A-Z]+$/))  {
```

Comment:
There are some currencies that have a ticker symbol with just two letters ([Aeternity](https://etherscan.io/token/0x5ca9a71b1d01849c0a95490cc00559717fcf0d1d) for example).

Reply:
I was not aware, and there are actually a handful that have 4 letters as well, so I'll relax this to be a length of 2-4.